docs/src/api/class-framelocator.md

```diff
@@ -0,0 +1,69 @@
+# class: FrameLocator
+
+FrameLocator represents a view to the `iframe` on the page. It captures the logic sufficient to retrieve the `iframe` and locate elements in that iframe. FrameLocator can be created with either [`method: Page.frameLocator`] or [`method: Locator.frameLocator`] method.
+
+```js
+const locator = page.frameLocator('iframe').locator('text=Submit');
+await locator.click();
+```
+
+```java
+Locator locator = page.frameLocator("iframe").locator("text=Submit");
+locator.click();
+```
+
+```python async
+locator = page.frame_locator("iframe").locator("text=Submit")
+await locator.click()
+```
+
+```python sync
+locator = page.frame_locator("text=Submit").locator("text=Submit")
+locator.click()
+```
+
+```csharp
+var locator = page.FrameLocator("iframe").Locator("text=Submit");
+await locator.ClickAsync();
+```
+
+## method: FrameLocator.frameLocator
+- returns: <[FrameLocator]>
+
+When working with iframes, you can create a frame locator that will enter the iframe and allow selecting elements
+in that iframe:
+
+```js
+const locator = page.frameLocator('iframe').locator('text=Submit');
```

Comment:
This should be something like `frameLocator('#my-frame').frameLocator('.preview-frame')`

Reply:
I don't think users would have a lot of nested frames, feels like me example is more relevant. I can use iframe#my-frame for clarity though.